telemetry_package/main.py

```diff
@@ -1,33 +1,145 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import re
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
+from rclpy.qos import QoSProfile, QoSReliabilityPolicy, QoSHistoryPolicy, QoSDurabilityPolicy, QoSLivelinessPolicy
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        super().__init__('telemetry_node')
+        qos_profile = QoSProfile( 
+            history=QoSHistoryPolicy.KEEP_LAST,
+            depth=1,
+            reliability=QoSReliabilityPolicy.BEST_EFFORT,
+            durability=QoSDurabilityPolicy.VOLATILE, 
+            liveliness = QoSLivelinessPolicy.AUTOMATIC
+        )
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists.')
+            raise FileExistsError
+            
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Accept the connection
+            self.client_sock, _ = self.server_sock.accept()
+            print('Connection accepted. Creating subscription')
+
+            # After connecting, create two subscriptions: one to the '/rosout' topic and one to the 'control' topic
+            self.rosout_subscription = self.create_subscription(
+                Log,
+                '/rosout',
+                self.rosout_callback, 
+                qos_profile = qos_profile)
+            
+            self.rosout_subscription
+            print('Created subscription to "/rosout".')
+
+            # Commenting this out for test purposes
+            # self.control_subscription = self.create_subscription(
+            #     Log,
+            #     'control',
+            #     self.control_subscription_callback, 
+            #     qos_profile = qos_profile)
```

Comment:
Is this still in progress? msg_type is probably not _Log_.

Reply:
This subscription would definitely be a nice to have.
But you're right, the message type is not `Log`.

Apart from that, we have the topic names defined globally: `TopicNames.Control`
See here for more details: https://docs.google.com/document/d/17nq2SS2DX0lrlWp5FjLQez1mpn0i5w_5APbjAiUk3j0/edit#heading=h.lfxu097txol